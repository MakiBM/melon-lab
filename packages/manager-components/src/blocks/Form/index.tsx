import React, { StatelessComponent } from 'react';

export interface ButtonProps {
  onSubmit?: any;
}

const Form: StatelessComponent<ButtonProps> = ({ children, onSubmit }) => {
  return (
    <form onSubmit={onSubmit} className="form">
      {children}
    </form>
  );
};

export default Form;
