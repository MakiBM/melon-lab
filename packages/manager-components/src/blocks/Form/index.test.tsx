import React from 'react';
import Form from './index';

describe('Form', () => {
  const defaultElement = <Form>Default form</Form>;

  it('should render correctly', () => {
    const wrapper = shallow(defaultElement);
    expect(wrapper).toMatchSnapshot();
  });

  it('should be selectable by class "form"', () => {
    expect(shallow(defaultElement).is('.form')).toBe(true);
  });

  it('should mount in a full DOM', () => {
    expect(mount(defaultElement).find('.form').length).toBe(1);
  });

  it('should render to static HTML', () => {
    expect(render(defaultElement).text()).toEqual('Default form');
  });
});
