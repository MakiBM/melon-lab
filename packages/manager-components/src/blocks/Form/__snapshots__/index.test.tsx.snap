// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should render correctly 1`] = `
<form
  className="jsx-__scopedHash form"
>
  <JSXStyle
    css="__scoped"
    styleId="__scopedHash"
  />
  Default form
</form>
`;
